### ISSUE TITLE
[wgsl-in] Handling types with '>>' like: 'ptr<uniform, array<i32, 12>>' (lexer '>>' vs. '> >' ?)

### ISSUE THREAD

**Issue by pjoe** [gfx-rs/naga#242]
The following doesn't work:
```
[[stage(vertex)]]
fn main() -> void {
  var x: ptr<uniform, array<i32, 12>>;
}
```

This gives the error: `Unexpected(ShiftOperation('>'))`.

Adding a space between the two '>' works though:
```
[[stage(vertex)]]
fn main() -> void {
  var x: ptr<uniform, array<i32, 12> >;
}
```

**Comment by pjoe** [comment]
FWIW: @Kangz pointed to llvm's handling of template params: https://github.com/llvm/llvm-project/blob/d1446017f3fdc2f6a9efba222008d20afa1e26cc/clang/lib/Parse/ParseTemplate.cpp#L492

**Comment by kvark** [comment]
Fixed by #476
